pkg/sfu/streamallocator/streamallocator.go

```diff
@@ -641,9 +641,14 @@ func (s *StreamAllocator) handleSignalEstimate(event *Event) {
 
 func (s *StreamAllocator) handleSignalPeriodicPing(event *Event) {
 	// finalize probe if necessary
-	isHandled, isSuccessful := s.probeController.MaybeFinalizeProbe()
+	trend, _ := s.channelObserver.GetTrend()
+	isHandled, isNotFailing, isGoalReached := s.probeController.MaybeFinalizeProbe(
+		s.channelObserver.HasEnoughEstimateSamples(),
+		trend,
+		s.channelObserver.GetLowestEstimate(),
+	)
 	if isHandled {
-		s.onProbeDone(isSuccessful)
+		s.onProbeDone(isNotFailing, isGoalReached)
```

Comment:
Sorry about the variable naming here. I changed `isSuccessful` -> `isNotFailing` because it is not really a successful (in the sense of attaining the goal) probe, but it is just not a failing probe. Exactly same meaning in code, but making reading the code harder (double negatives). Let me know if you can think of a better name.